airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -67,7 +64,13 @@ def _validate_and_transform(self, config: Mapping[str, Any]):
         config.setdefault("action_breakdowns_allow_empty", False)
         if config.get("end_date") == "":
             config.pop("end_date")
+
+        # set default start_date to 2 years ago
+        if not config.get("start_date"):
+            config["start_date"] = pendulum.now().add(years=-2)
```

Comment:
Better to specify `default_start_date` property and return calculated value  `pendulum.now(tz="UTC").subtract(years=2).format("YYYY-MM-DDTHH:mm:ss[Z]")`

And it can be look like that:
```
@property
def default_start_date(self) -> str:
    return pendulum.now(tz="UTC").subtract(years=2).format("YYYY-MM-DDTHH:mm:ss[Z]")

def _validate_and_transform(self, config: Mapping[str, Any]):
    ...
    config.setdefault("start_date", self.default_start_date)  # set default start_date to 2 years ago if not specified
    ...
```

Also I didn't find that the Facebook Marketing API requires a date range parameter. Can you confirm this?

Reply:
update usage of optional start_date